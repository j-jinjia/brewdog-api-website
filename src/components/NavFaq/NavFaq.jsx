import React from "react";
import "./NavFaq.scss";
import cross from "../../assets/images/white-cross.png";
const NavFaq = ({ handleFaqClick }) => {
  return (
    <div className="help">
      <img
        className="help__cross"
        src={cross}
        alt="white-cross"
        onClick={handleFaqClick}
      />
      <h3 className="help__title">Need help?</h3>
      <p className="help__subtitle">Let me show you around!</p>
    </div>
  );
};

export default NavFaq;
