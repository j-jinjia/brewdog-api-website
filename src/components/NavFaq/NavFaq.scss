.help{
    position:absolute;
    background-color: pink;
    top:0;
    bottom: 0;
    left: 0;
    right:0;
    z-index: 5;
    &__cross{
        cursor: pointer;
    }
}