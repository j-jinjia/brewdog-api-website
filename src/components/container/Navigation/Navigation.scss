.navigation{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    background-color: grey;
    .navigation__logo{
        width:50px;
        height: auto;

    }
    .navigation__faq{
        cursor: pointer;
    }
}