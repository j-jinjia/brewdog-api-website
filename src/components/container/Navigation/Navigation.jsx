import "./Navigation.scss";

import icon from "../../../assets/images/brewdog-icon.png";
import NavFaq from "../../NavFaq/NavFaq";

import { React, useState } from "react";

const Navigation = () => {
  const [showFaq, setShowFaq] = useState(false);
  const handleFaqClick = () => {
    setShowFaq(!showFaq);
  };
  return (
    <article className="navigation">
      {showFaq && <NavFaq handleFaqClick={handleFaqClick} />}
      <img src={icon} alt="brewdog-icon" className="navigation__logo" />
      <h1>BREWDOG</h1>
      <h2 className="navigation__faq" onClick={handleFaqClick} tabIndex="0 ">
        HELP
      </h2>
    </article>
  );
};

export default Navigation;
