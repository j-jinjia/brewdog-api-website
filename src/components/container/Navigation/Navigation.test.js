import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NavFaq from "../../NavFaq/NavFaq";
import Navigation from "./Navigation"


it("should match beer card component snapshot", ()=>{
    const {container} = render (<Navigation />)
    expect(container).toMatchSnapshot();

})

it("should render the navigation bar", ()=>{
   render(<Navigation/>)
   const navBar = screen.getByRole("article")
   expect(navBar).toBeInTheDocument();

})

it("help should appear in navigation bar", ()=>{
    render(<Navigation/>)
    const help = screen.getByText("HELP");
    expect(help).toBeInTheDocument();

})