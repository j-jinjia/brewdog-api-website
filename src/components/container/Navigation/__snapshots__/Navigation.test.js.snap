// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match beer card component snapshot 1`] = `
<div>
  <article
    class="navigation"
  >
    <img
      alt="brewdog-icon"
      class="navigation__logo"
      src="brewdog-icon.png"
    />
    <h1>
      BREWDOG
    </h1>
    <h2
      class="navigation__faq"
      tabindex="0 "
    >
      HELP
    </h2>
  </article>
</div>
`;
